gr-zeromq/lib/base_impl.h

```diff
@@ -23,8 +23,12 @@ class base_impl : public virtual gr::sync_block
     base_impl(int type,
               size_t itemsize,
               size_t vlen,
+              char* address,
               int timeout,
               bool pass_tags,
+              int hwm,   // high watermark
+              bool sink, // is sink, else source
```

Comment:
The interface problem here is that there is insufficient semantic meaning for this argument. The bind/connect flag is better, because ZMQ sockets are agnostic to bind vs connect (with the exception that binding more than once is an error, but that's enforced by the OS not the library).

What does sink vs not sink mean? Nothing to users of the class; it's a lower-level implementation detail of which socket option to set. These kinds of details should not be promoted to interfaces.